### PR TITLE
MOB-1479 - Added support of  XMTP service 

### PR DESCRIPTION
**Pull request by Oleg-Pecheneg**

### CODE REVIEW

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/ChatViewPresenter.swift [hunk 1]
```diff
@@ -184,8 +187,15 @@ extension ChatViewPresenter: MessagingServiceListener {
                    updatedMessage.chatId == chat.id,
                    let i = self.messages.firstIndex(where: { $0.id == updatedMessage.id }) {
                     await newMessage.prepareToDisplay()
-                    self.messages[i] = newMessage
-                    await addMessages([])
+                    self.messages.remove(at: i)
+//                    self.messages[i] = newMessage
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/ChatViewPresenter.swift:191]: dead?

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/Views/ChatEmptyView.swift [hunk 2]
```diff
@@ -128,6 +129,8 @@ extension ChatEmptyView {
                 return String.Constants.messagingChatEmptyEncryptedMessage.localized()
             case .chatUnEncrypted:
                 return String.Constants.messagingChatEmptyUnencryptedMessage.localized()
+            case .cantContact:
+                return "User is not yet on the XMTP network" // TODO: - Localize
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/Views/ChatEmptyView.swift:133]: todo

File: unstoppable-ios-app/domains-manager-ios/Services/MessagingService/SubServices/Descryption/XMTPMessagingContentDecrypterService.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    private let xmtpHelper = XMTPServiceHelper()
+
+    func isMessageEncrypted(serviceMetadata: Data?) -> Bool {
+        true
+    }
+    
+    func encryptText(_ text: String) throws -> String { text }
+
+    func decryptText(_ text: String, with serviceMetadata: Data?, wallet: String) throws -> String {
+//        guard let serviceMetadata,
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/MessagingService/SubServices/Descryption/XMTPMessagingContentDecrypterService.swift:22]: dead?

File: unstoppable-ios-app/domains-manager-ios/Services/MessagingService/XMTP/XMTPEntitiesTransformer.swift [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+        
+        
+        let displayInfo = MessagingChatMessageDisplayInfo(id: id,
+                                                          chatId: chat.displayInfo.id,
+                                                          userId: userId,
+                                                          senderType: sender,
+                                                          time: time,
+                                                          type: type,
+                                                          isRead: isRead,
+                                                          isFirstInChat: false, // TODO: - Set this property
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/MessagingService/XMTP/XMTPEntitiesTransformer.swift:112]: it's set, isn't it?

**Review by rommex — COMMENTED** [review]
with minor suggestions

File: unstoppable-ios-app/domains-manager-ios/Entities/Storages/Keychain/KeychainLayer.swift [hunk 5]
```diff
@@ -117,6 +118,27 @@ extension PrivateKeyStorage {
         try? valet.removeObject(forKey: pubKeyHex.normalized)
     }
 }
+extension PrivateKeyStorage {
+    func store(data: Data, for key: String) {
+        do {
+            try valet.setObject(data, forKey: key)
+            Debugger.printInfo("Stored data to keychain: \(key)")
+        } catch {
+            Debugger.printFailure("Failed to store data keychain: \(key) with error \(error.localizedDescription)", critical: true)
```
[the remark this answers is not in the document]
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Entities/Storages/Keychain/KeychainLayer.swift:127]: Fixed typo

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/Views/ChatEmptyView.swift (hunk 2 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/Views/ChatEmptyView.swift:133]: Localized

File: unstoppable-ios-app/domains-manager-ios/Services/MessagingService/SubServices/Descryption/XMTPMessagingContentDecrypterService.swift (hunk 3 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Services/MessagingService/SubServices/Descryption/XMTPMessagingContentDecrypterService.swift:22]: Removed

File: unstoppable-ios-app/domains-manager-ios/Services/MessagingService/XMTP/XMTPEntitiesTransformer.swift (hunk 4 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Services/MessagingService/XMTP/XMTPEntitiesTransformer.swift:112]: Correct. Removed TODO

**Comment by Oleg-Pecheneg** [comment]
@rommex Mentioned issues is resolved, please proceed with review

**Review by rommex — APPROVED** [review]

**Pull request merged**